blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java

```diff
@@ -74,7 +75,12 @@ public String getDisplayName() {
 
     @Override
     public String getDisplayDescription() {
-        return ArgumentsAction.getStepArgumentsAsString(node.getNode());
+        String displayDescription = ArgumentsAction.getStepArgumentsAsString(node.getNode());
+        if (displayDescription != null) {
+            // JENKINS-45099 Remove any control characters that may have found their way out of a script
+            displayDescription = CharMatcher.JAVA_ISO_CONTROL.and(CharMatcher.anyOf("\r\n\t").negate()).removeFrom(displayDescription);
```

Comment:
@i386 Code looks good, if I read it correctly, it filters out ISO controls chars (not json friendly) and line/tab breaks. We should keep it inside commons package, maybe inside Utils.java as generic code, could be used somewhere else if needed, perhaps something along this?

```
public String sanitizeJsonString(@Nonnull String value, @Nullable String additionalFilter){
        
        CharMatcher charMatcher = CharMatcher.JAVA_ISO_CONTROL;
        if(StringUtils.isNotBlank(additionalFilter)){
            charMatcher = charMatcher.and(CharMatcher.anyOf(additionalFilter).negate());
        }
        return charMatcher.removeFrom(value);
    }
```